pkg/kubectl/resource/builder.go

```diff
@@ -92,14 +92,20 @@ func (b *Builder) Schema(schema validation.Schema) *Builder {
 	return b
 }
 
+type FilenameParamOptions struct {
```

Comment:
I like structuring the flags with this struct and binding the flags to it.  It makes it easier for composition and consumption.